functors_task_1/student/include/impl.h

```diff
@@ -16,16 +25,23 @@ struct BeerOrganizer
  *
  * @note Only Corona and HoeGaarden are expensive
  */
-bool isExpensiveBeer(/**???*/)
+bool isExpensiveBeer(const BeerBrand beerBrand)
 {
+    return (BeerBrand::Corona==beerBrand || BeerBrand::HoeGaarden==beerBrand);
 }
 
 /**
  * @todo Implement lambda beer country equality comparator
  * @return true if beer county is the same, false otherwise
  */
-auto sameCountry = [](/**???*/)
+auto sameCountry = [](const BeerBrand lhs,const BeerBrand rhs)
 {
+    const auto lCountry= getBeerCountry(lhs);
+    const auto rCountry= getBeerCountry(rhs);
+    if(Country::Unknown==lCountry){
+        return false;
+    }
+    return lCountry==rCountry;
```

Comment:
I suppose you can perform check inline here:
```suggestion
    return getBeerCountry(lhs) == getBeerCountry(rhs)
```

Reply:
no becouse of Country::Unknown

Reply:
Still don't get why do you think that Unknown is a special case